tools/Status/ExecutableStatus/ExecutableStatus.py

```diff
@@ -190,6 +202,51 @@ def format(self, field, value):
             return f"{value:g}"
         raise ValueError
 
+    def render_time_steps(self, input_file: dict, reduction_files: list):
+        import plotly.express as px
+        import streamlit as st
+
+        observe_time_event = find_event("ObserveTimeStep", input_file)
+        if observe_time_event is None:
+            st.warning("No 'ObserveTimeStep' event found in input file.")
+            return
+        subfile_name = observe_time_event["SubfileName"] + ".dat"
+        logger.debug(f"Reading time steps from subfile: '{subfile_name}'")
+
+        def get_time_steps(reductions_file):
+            with h5py.File(reductions_file, "r") as open_h5file:
+                return to_dataframe(open_h5file[subfile_name]).set_index("Time")
+
+        st.subheader("Time steps")
+        time_steps = pd.concat(map(get_time_steps, reduction_files))
+        fig = px.line(
+            time_steps,
+            y=[
+                "Effective time step",
+                "Minimum time step",
+                "Maximum time step",
+                "Slab size",
+            ],
+            log_y=True,
+        )
+        fig.update_layout(
+            legend=dict(
+                title=None,
+                orientation="h",
+                yanchor="bottom",
+                y=1,
+                xanchor="left",
+                x=0,
+            ),
+            yaxis=dict(exponentformat="e", title=None),
+        )
+        st.plotly_chart(fig)
+
+    def render_dashboard(self, job: dict, input_file: dict):
+        return self.render_time_steps(
+            input_file, list_reduction_files(job, input_file)
+        )
+
 
 class EllipticStatus(ExecutableStatus):
```

Comment:
I think you've also added a dashboard for the elliptic one too since you opened this. Wanna add that? I think the only thing is figuring out how to show the latest control step and not just the first one.

Reply:
I added the elliptic dashboard. It only shows the convergence for the first control step atm, and also plots the convergence of the control parameter. I'll probably combine these somehow, but haven't decided how yet.